libraries/chain/controller.cpp

```diff
@@ -3999,15 +3986,86 @@ struct controller_impl {
       bsp->verify_qc(qc_proof);
    }
 
+   // Verify QC block extension and finality block header extension on a Legacy
+   // or Transition block.
+   void verify_legacy_block_exts( const signed_block_ptr& b, const block_header_state_legacy& prev ) {
+      uint32_t block_num = b->block_num();
+
+      auto block_exts = b->validate_and_extract_extensions();
+      auto qc_ext_id = quorum_certificate_extension::extension_id();
+      bool qc_extension_present = block_exts.count(qc_ext_id) != 0;
+      EOS_ASSERT( !qc_extension_present, invalid_qc_claim,
+                  "Legacy or Transition block #${b} includes a QC block extension",
+                  ("b", block_num) );
+
+      EOS_ASSERT( !b->is_proper_svnn_block(), invalid_qc_claim,
```

Comment:
wrong exception type `invalid_qc_claim` used here and in some following checks.

Reply:
I used `block_validate_exception` for some of them. Kevin suggested using `invalid_qc_claim` for all the checks here. It makes sense as they are all related QC.

Reply:
I agree with Greg. 

I think this particular `EOS_ASSERT` should use `block_validate_exception` since it does not actually have to do with QC claims in the block.